### DISCUSSION TITLE
Use early_stopping or not?

### DISCUSSION THREAD

**Discussion by guanjen375** [kyleskom/NBA-Machine-Learning-Sports-Betting#226] [Ideas]
I have noticed the XGB training is as follows:

    for x in tqdm(range(100)):
        x_train, x_test, y_train, y_test = train_test_split(data, margin, test_size=.1)

        train = xgb.DMatrix(x_train, label=y_train)
        test = xgb.DMatrix(x_test, label=y_test)

        param = {
            'max_depth': 2,
            'eta': 0.01,
            'objective': 'multi:softprob',
            'num_class': 2
        }
        epochs = 500

        model = xgb.train(param, train, epochs)
        predictions = model.predict(test)
        y = []

        for z in predictions:
            y.append(np.argmax(z))

        acc = round(accuracy_score(y_test, y)*100, 1)

I have also noticed some people do not use the forloop to train the models:

    x_train, x_test, y_train, y_test = train_test_split(data, margin, test_size=.1)

    train = xgb.DMatrix(x_train, label=y_train)
    test = xgb.DMatrix(x_test, label=y_test)

    param = {
        'max_depth': 2,
        'eta': 0.01,
        'objective': 'multi:softprob',
        'num_class': 2
    }
    epochs = 500
    eval_set = [(x_test,y_test)]

    model = xgb.train(param, train, epochs, early_stopping_rounds=10, eval_set = eval_set )

    predictions = model.predict(test)
    y = []

    for z in predictions:
        y.append(np.argmax(z))

    acc = round(accuracy_score(y_test, y)*100, 1)

Which one do you prefer? Or there is another better way to train the model?
